OpenRA.Mods.Common/Traits/Production/ProductionByDelivery.cs

```diff
@@ -0,0 +1,198 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Deliver produced units from outside the map.")]
+	public class ProductionByDeliveryInfo : ProductionInfo
+	{
+		[Desc("Delivering actor name.")]
```

Comment:
I like the idea of instead fixing this to support helicopters (but then still fixing this to mention it needs to be Helicopter of Plane).  Helicopter flight physics would fit the d2k delivery better than Plane.